### PR TITLE
Updated CPU manager configuration details.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
- Use existing node allocatable settings instead of a new one.

cc @sjenning @flyingcougar @balajismaniam

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
Quick link to rendered section: https://github.com/ConnorDoyle/community/blob/cpu-manager-proposal-config/contributors/design-proposals/cpu-manager.md#configuring-the-cpu-manager

**Comment by sjenning** [comment]
I like this.  Should this be `ceiling` rather than `floor` though?  If Allocatable is 3500m, the expected result in my mind is that 3 cores can be pinned and 1 core can not i.e. is permanently in the shared pool.  However the cpumanager still needs to manage the fractional part of that 4th core even if the scheduler will not assign 4 pinned cores worth of pods to the node.

If we use floor, and we already have 3 cores of pinned pods on the node, the cpumanger with think the shared pool is depleted, when really the 4th core should be in the shared pool, right?

File: contributors/design-proposals/cpu-manager.md [hunk 1]
```diff
@@ -134,12 +134,24 @@ type CPUSet map[int]struct{} // set operations and parsing/formatting helpers
...
+#### Configuring the CPU Manager
+
 Kubernetes will ship with three CPU manager policies. Only one policy is
 active at a time on a given node, chosen by the operator via Kubelet
 configuration. The three policies are **no-op**, **static** and **dynamic**.
 
-Operators can set the active CPU manager policy through a new Kubelet
-configuration setting `--cpu-manager-policy`.
+The active CPU manager policy is set through a new Kubelet
+configuration value `--cpu-manager-policy`.
```
Comment (balajismaniam) [inline contributors/design-proposals/cpu-manager.md]: May be we can specify default too? (set to `no-op`.)

File: contributors/design-proposals/cpu-manager.md [hunk 2]
```diff
@@ -134,12 +134,24 @@ type CPUSet map[int]struct{} // set operations and parsing/formatting helpers
...
+
+The number of CPUs that pods may run on is set using the existing
+node-allocatable configuration settings. See the [node allocatable proposal
```
Comment (balajismaniam) [inline contributors/design-proposals/cpu-manager.md]: Use this link https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node-allocatable.md#phase-2---enforce-allocatable-on-pods for node-allocatable.

File: contributors/design-proposals/cpu-manager.md [hunk 3]
```diff
@@ -134,12 +134,24 @@ type CPUSet map[int]struct{} // set operations and parsing/formatting helpers
...
+document][node-allocatable] for details. The CPU manager will claim
+`floor(node.status.allocatable.cpu)` as the number of CPUs available to assign
+to pods, starting from the highest-numbered physical core and descending
+topologically.
+
+Operator documentation will be updated to explain how to configure the
+system to use the low-numbered physical cores for kube and system slices.
```
Comment (balajismaniam) [inline contributors/design-proposals/cpu-manager.md]: s/kube/kubereserved, s/system/systemreserved

File: contributors/design-proposals/cpu-manager.md (hunk 1 above)
Comment (ConnorDoyle) [reply contributors/design-proposals/cpu-manager.md]: Yes, good idea.

File: contributors/design-proposals/cpu-manager.md (hunk 2 above)
Comment (ConnorDoyle) [reply contributors/design-proposals/cpu-manager.md]: OK, will update.

**Comment by ConnorDoyle** [comment]
@sjenning 

> Should this be ceiling rather than floor though?

After more thought, ceiling seems more correct than floor. However, another option could be to error if Allocatable.cpu isn't an integer. In trying to handle ceiling(3500m) as in your example, we would end up with a CPU in multiple scheduling domains which brings other issues. At least, I would recommend making the best practice to configure an integer-valued reservation for cpu.

Anyone in favor of error vs warn + document?

**Comment by sjenning** [comment]
> At least, I would recommend making the best practice to configure an integer-valued reservation for cpu.

I thought this as well.  I think it's a good idea to advise (require?) CPU kube/system-reserved be an integer value when using the static cpu manager.

@derekwaynecarr any thoughts?

**Comment by ConnorDoyle** [comment]
@sjenning updated. We can add the warning / error part once we make a decision on it. Think it's safe to say for now that integer allocatable.cpu is "recommended".

**Pull request merged**